execution_workflows/APAtrap/modules/predict_apa.nf

```diff
@@ -0,0 +1,26 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+/*
+    Run the second step of APAtrap: predictAPA to infer all potential
+    APA sites and estimate their corresponding usages.
+*/
+process PREDICT_APA {
+        publishDir "${params.outdir}/apatrap", mode: params.publish_dir_mode
+        container "docker.io/apaeval/apatrap:latest"
+
+        input:
+        tuple path(reads_bedgraph_file), path(predict_apa_input)
+
+        output:
+        path "$predict_apa_output", emit: ch_de_apa_input
+
+        script:
+        predict_apa_output = "de_apa_input.txt"
+        """
+        predictAPA -i $reads_bedgraph_file -g 1 -n 1 -u $predict_apa_input -o $predict_apa_output
```

Comment:
Again I don't think parameter values should be hardcoded. For differential usage benchmarks we will definitely have more than one group (`-g 1`) and > 1 sample per group (`-n 1`). As with `identifyDistal3UTR` I think optional parameters should be specified in call and defined in a config file, even if they are set to defaults for now.

```
-d  minimum degree of coverage variation between two adjcent APA sites, >0 and <1, default is 0.2.
-c  minimum average coverage required for each 3'UTR, >=10, default is 20.
-a  minimum distance between the predicted APA sites, >=20, default is 100.
-w  window size used to scan the profile, >=20, default is 50.
```